File: TransCoder_tokenized_test_set_functions/java/SORT_STRING_ACCORDING_ORDER_DEFINED_ANOTHER_STRING.java
static void sortByPattern ( char [ ] str, char [ ] pat ) {
  int count [ ] = new int [ MAX_CHAR ];
  for ( int i = 0;
  i < str . length;
  i ++ ) {
    count [ str [ i ] - 'a' ] ++;
  }
  int index = 0;
  for ( int i = 0;
  i < pat . length;
  i ++ ) {
    for ( int j = 0;
    j < count [ pat [ i ] - 'a' ];
    j ++ ) {
      str [ index ++ ] = pat [ i ];
    }
  }
}

File: TransCoder_tokenized_test_set_functions/cpp/SORT_STRING_ACCORDING_ORDER_DEFINED_ANOTHER_STRING.cpp
void sortByPattern ( string & str, string pat ) {
  int count [ MAX_CHAR ] = {
    0 };
    for ( int i = 0;
    i < str . length ( );
    i ++ ) count [ str [ i ] - 'a' ] ++;
    int index = 0;
    for ( int i = 0;
    i < pat . length ( );
    i ++ ) for ( int j = 0;
    j < count [ pat [ i ] - 'a' ];
    j ++ ) str [ index ++ ] = pat [ i ];
  }